### DISCUSSION TITLE
Nested array property and checkboxes

### DISCUSSION THREAD

**Discussion by solarsilk** [livewire/livewire#5496] [1. Help] [answered]
I have a model property which is a JSON column with 100+ properties, and some of the values of those nested JSON properties are arrays. eg:
```
$model->json_column['nested_prop'] // value is an array [0,1,2]
```
I have groups of checkboxes for each of these properties, like this:
```
<input type="checkbox" wire:model.defer="doc.content.recommedations" value="0">
<input type="checkbox" wire:model.defer="doc.content.recommedations" value="1">
<input type="checkbox" wire:model.defer="doc.content.recommedations" value="2">
```
Livewire doesn't deal with these as I'd hoped. Seems like it's saving just one value, not the array of values.

In [this video on the Livewire site](https://laravel-livewire.com/screencasts/form-checkboxes), they point out that if a public property is/should be an array, it needs to be initialized like:
```
public $prop = [];
```
So I probably need to initialize those JSON properties to empty arrays somehow. But how do I do that, given that the `$doc->content` that's passed in to the component has different properties depending on the doc so I can't hard-code them (and there are lots). Or is there some other approach I should use? Thanks for any help.

**Comment by solarsilk** [comment, marked as answer]
I tracked down the issue - it was a problem with a related thing I was doing in Alpine with the checkboxes that was messing it up somehow. After I fixed that, it's working as expected.

**Comment by theChaosCoder** [comment]
How did you fix it? I have a similar issue, but I'm not using alpine:

This works
`public $test= [];`

```
<input type="checkbox" wire:model="test" value="123">
<input type="checkbox" wire:model="test" value="abc">
```

-------------
But this does not
```
public $fields = [
        1 => [
            'test' => [],
        ],
],
```

```
<input type="checkbox" wire:model="fields.1.test" value="123">
<input type="checkbox" wire:model="fields.1.test" value="abc">
```

I'm using the latest Livewire version v2.11.2

**Reply by solarsilk** [reply]
Your controller code looks like it should work. 

What I do is loop through all the properties (which are all arrays) of the model and set them to either their current value or `[]` like this:
```
foreach ($questions as $question) {
    $this->doc->content[$question->name] = $this->doc->content[$question->name] ?? [];
}
```

You can try adding a `name` attribute to all the checkboxes in a group. Not sure if that will help but you can try that.
